azure/Guidelines.md

```diff
@@ -0,0 +1,864 @@
+
+*NOTE: For an existing GA'd service, don't change/break its existing API; instead, leverage these concepts for future APIs while prioritizing consistency within your existing service.*
+
+### Prescriptive Guidance  
+This document offers prescriptive guidance labeled as follows:
+
+> :white_check_mark: **DO** adopt this pattern. If you feel you need an exception, contact the Azure HTTP/REST Stewardship Board <b>prior</b> to implementation.
+ 
+> :ballot_box_with_check: **YOU SHOULD** adopt this pattern. If not following this advice, you MUST disclose your reason during the Azure HTTP/REST Stewardship Board review.
+ 
+> :heavy_check_mark: **YOU MAY** consider this pattern if appropriate to your situation. No notification to the Azure HTTP/REST Stewardship Board is required.
+ 
+> :warning: **YOU SHOULD NOT** adopt this pattern. If not following this advice, you MUST disclose your reason during the Azure HTTP/REST Stewardship Board review.
+ 
+> :no_entry: **DO NOT** adopt this pattern. If you feel you need an exception, contact the Azure HTTP/REST Stewardship Board <b>prior</b> to implementation.
+ 
+*If you feel you need an exception, or need clarity based on your situation, please contact the Azure HTTP/REST Stewardship Board <b>prior</b> to release of your API.*
+
+## Building Blocks: HTTP, REST, & JSON
+The Microsoft Azure Cloud platform exposes its APIs through the core building blocks of the Internet; namely HTTP, REST, and JSON. This section provides you with a general understanding of how these technologies should be applied when creating your service.
+
+### HTTP
+Azure services must adhere to the HTTP specification, [RFC7231](https://tools.ietf.org/html/rfc7231). This section further refines and constrains how service implementors should apply the constructs defined in the HTTP specification. It is therefore, important that you have a firm understanding of the following concepts:
+* [Uniform Resource Locators (URLs)](URLS)
+* HTTP Methods
+* Request & Response Headers
+* Bodies
+
+### Uniform Resource Locators (URLs)
+
+A Uniform Resource Locator (URL) is how developers access the resources of your service. Ultimately, URLs are how developers form a cognitive model of your service's resources.
+
+> :white_check_mark: **DO** use this URL pattern:
+```text
+https://<service>.<cloud>/<tenant>/<service-root>/<resource-collection>/<resource-id>/
+```
+
+Where:
+ | Field | Description
+ | - | - |
+ | service | Name of the service (ex: blobstore, servicebus, directory, or management)
+ | cloud | Cloud domain name (see Azure CLI's "az cloud list")<p><table><tr><td><b>Cloud</td><td><b>Domain</td></tr><tr><td>Public</td><td>azure.net</td></tr><tr><td>US Government</td><td>usgovcloudapi.net</td></tr><tr><td>China</td><td>chinacloudapi.cn</td></tr><tr><td>German</td><td>cloudapi.de</td></tr></table>
```

Comment:
perhaps note that black forest is being deprecated. 
we should also include Fairfax and Mooncake aliases for usgov and china